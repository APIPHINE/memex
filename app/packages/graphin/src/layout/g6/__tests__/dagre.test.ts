import { Data } from '../../../types';
import DagreLayout from '../dagre';
import json from '../../basic/__tests__/__mock__/concentric.input.json';

const defaultOptions = {
  /** 节点大小 */
  nodeSize: [50, 50] as [number, number],
  /**  节点水平间距(px) */
  nodesep: 12,
  /** 每一层节点之间间距 */
  ranksep: 50,
  /** 防止位置 */
  align: 'UL',
};

const data: Data = json as any; // eslint-disable-line

describe('Dagre Layout', () => {
  it('Should return result that matches snapshot', () => {
    expect(DagreLayout(data, defaultOptions)).toMatchSnapshot();
  });
});
