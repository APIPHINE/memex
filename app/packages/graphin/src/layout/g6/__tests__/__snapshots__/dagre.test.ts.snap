// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dagre Layout Should return result that matches snapshot 1`] = `
Object {
  "edges": Array [
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-0",
        "target": "node-0",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "loop",
      "source": "node-0",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-0",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-0",
        "target": "node-1",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-0",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-1",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-0",
        "target": "node-2",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-0",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-2",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-0",
        "target": "node-3",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-0",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-3",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-0",
        "target": "node-4",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-0",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-4",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-0",
        "target": "node-5",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-0",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-5",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-0",
        "target": "node-6",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-0",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-6",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-0",
        "target": "node-7",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-0",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-7",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-0",
        "target": "node-8",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-0",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-8",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-1",
        "target": "node-0",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-1",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-0",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-2",
        "target": "node-0",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-2",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-0",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-3",
        "target": "node-0",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-3",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-0",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-4",
        "target": "node-0",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-4",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-0",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-5",
        "target": "node-0",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-5",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-0",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-6",
        "target": "node-0",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-6",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-0",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-7",
        "target": "node-0",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-7",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-0",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-8",
        "target": "node-0",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-8",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-0",
    },
    Object {
      "data": Object {
        "properties": Array [],
        "source": "node-9",
        "target": "node-0",
      },
      "loopCfg": Object {
        "dist": 20,
        "position": "top",
      },
      "shape": "LineEdge",
      "source": "node-9",
      "style": Object {
        "cursor": "pointer",
        "endArrow": true,
        "lineAppendWidth": 10,
        "lineWidth": 1.5,
        "stroke": "#ddd",
      },
      "target": "node-0",
    },
  ],
  "nodes": Array [
    Object {
      "data": Object {
        "id": "node-0",
        "label": "徐臣莺某某公司(node-0)",
        "properties": Array [
          Object {
            "code": "name",
            "name": "公司名称",
            "value": "徐臣莺某某公司",
          },
          Object {
            "code": "companyType",
            "name": "类型",
            "value": "非上市公司",
          },
          Object {
            "code": "industry",
            "name": "行业",
            "value": "传统行业",
          },
          Object {
            "code": "phone",
            "name": "电话",
            "value": "202372158",
          },
          Object {
            "code": "USCC",
            "name": "统一社会信用代码",
            "value": "9357810591505709",
          },
          Object {
            "code": "registered capital",
            "name": "注册资本",
            "value": "27991.8147(万元)",
          },
          Object {
            "code": "number of participants",
            "name": "参保人数",
            "value": 54,
          },
          Object {
            "code": "register prov",
            "name": "注册省份",
            "value": "吉林省",
          },
        ],
        "type": "company",
      },
      "id": "node-0",
      "shape": "CircleNode",
      "style": Object {
        "dark": "#eee",
        "fontColor": "#3b3b3b",
        "fontSize": 12,
        "nodeSize": 20,
        "primaryColor": "#515151",
      },
      "x": 37,
      "y": 275,
    },
    Object {
      "data": Object {
        "id": "node-1",
        "label": "郭精某某公司(node-1)",
        "properties": Array [
          Object {
            "code": "name",
            "name": "公司名称",
            "value": "郭精某某公司",
          },
          Object {
            "code": "companyType",
            "name": "类型",
            "value": "非上市公司",
          },
          Object {
            "code": "industry",
            "name": "行业",
            "value": "互联网行业",
          },
          Object {
            "code": "phone",
            "name": "电话",
            "value": "673928692",
          },
          Object {
            "code": "USCC",
            "name": "统一社会信用代码",
            "value": "3703448383706587",
          },
          Object {
            "code": "registered capital",
            "name": "注册资本",
            "value": "39244.7661(万元)",
          },
          Object {
            "code": "number of participants",
            "name": "参保人数",
            "value": 55,
          },
          Object {
            "code": "register prov",
            "name": "注册省份",
            "value": "福建省",
          },
        ],
        "type": "company",
      },
      "id": "node-1",
      "shape": "CircleNode",
      "style": Object {
        "dark": "#eee",
        "fontColor": "#3b3b3b",
        "fontSize": 12,
        "nodeSize": 20,
        "primaryColor": "#515151",
      },
      "x": 37,
      "y": 475,
    },
    Object {
      "data": Object {
        "id": "node-2",
        "label": "暨刻瑜某某公司(node-2)",
        "properties": Array [
          Object {
            "code": "name",
            "name": "公司名称",
            "value": "暨刻瑜某某公司",
          },
          Object {
            "code": "companyType",
            "name": "类型",
            "value": "上市公司",
          },
          Object {
            "code": "industry",
            "name": "行业",
            "value": "传统行业",
          },
          Object {
            "code": "phone",
            "name": "电话",
            "value": "057608991",
          },
          Object {
            "code": "USCC",
            "name": "统一社会信用代码",
            "value": "8148209249716394",
          },
          Object {
            "code": "registered capital",
            "name": "注册资本",
            "value": "50526.7307(万元)",
          },
          Object {
            "code": "number of participants",
            "name": "参保人数",
            "value": 69,
          },
          Object {
            "code": "register prov",
            "name": "注册省份",
            "value": "福建省",
          },
        ],
        "type": "company",
      },
      "id": "node-2",
      "shape": "CircleNode",
      "style": Object {
        "dark": "#eee",
        "fontColor": "#3b3b3b",
        "fontSize": 12,
        "nodeSize": 20,
        "primaryColor": "#515151",
      },
      "x": 123,
      "y": 475,
    },
    Object {
      "data": Object {
        "id": "node-3",
        "label": "鲁涤媚某某公司(node-3)",
        "properties": Array [
          Object {
            "code": "name",
            "name": "公司名称",
            "value": "鲁涤媚某某公司",
          },
          Object {
            "code": "companyType",
            "name": "类型",
            "value": "上市公司",
          },
          Object {
            "code": "industry",
            "name": "行业",
            "value": "互联网行业",
          },
          Object {
            "code": "phone",
            "name": "电话",
            "value": "026585762",
          },
          Object {
            "code": "USCC",
            "name": "统一社会信用代码",
            "value": "6397905239447341",
          },
          Object {
            "code": "registered capital",
            "name": "注册资本",
            "value": "83683.0498(万元)",
          },
          Object {
            "code": "number of participants",
            "name": "参保人数",
            "value": 98,
          },
          Object {
            "code": "register prov",
            "name": "注册省份",
            "value": "重庆市",
          },
        ],
        "type": "company",
      },
      "id": "node-3",
      "shape": "CircleNode",
      "style": Object {
        "dark": "#eee",
        "fontColor": "#3b3b3b",
        "fontSize": 12,
        "nodeSize": 20,
        "primaryColor": "#515151",
      },
      "x": 209,
      "y": 475,
    },
    Object {
      "data": Object {
        "id": "node-4",
        "label": "宗陂拆某某公司(node-4)",
        "properties": Array [
          Object {
            "code": "name",
            "name": "公司名称",
            "value": "宗陂拆某某公司",
          },
          Object {
            "code": "companyType",
            "name": "类型",
            "value": "非上市公司",
          },
          Object {
            "code": "industry",
            "name": "行业",
            "value": "传统行业",
          },
          Object {
            "code": "phone",
            "name": "电话",
            "value": "014216010",
          },
          Object {
            "code": "USCC",
            "name": "统一社会信用代码",
            "value": "023036598552708343",
          },
          Object {
            "code": "registered capital",
            "name": "注册资本",
            "value": "67834.7548(万元)",
          },
          Object {
            "code": "number of participants",
            "name": "参保人数",
            "value": 69,
          },
          Object {
            "code": "register prov",
            "name": "注册省份",
            "value": "河北省",
          },
        ],
        "type": "company",
      },
      "id": "node-4",
      "shape": "CircleNode",
      "style": Object {
        "dark": "#eee",
        "fontColor": "#3b3b3b",
        "fontSize": 12,
        "nodeSize": 20,
        "primaryColor": "#515151",
      },
      "x": 295,
      "y": 475,
    },
    Object {
      "data": Object {
        "id": "node-5",
        "label": "牛城某某公司(node-5)",
        "properties": Array [
          Object {
            "code": "name",
            "name": "公司名称",
            "value": "牛城某某公司",
          },
          Object {
            "code": "companyType",
            "name": "类型",
            "value": "非上市公司",
          },
          Object {
            "code": "industry",
            "name": "行业",
            "value": "互联网行业",
          },
          Object {
            "code": "phone",
            "name": "电话",
            "value": "958535499",
          },
          Object {
            "code": "USCC",
            "name": "统一社会信用代码",
            "value": "8552263219031175",
          },
          Object {
            "code": "registered capital",
            "name": "注册资本",
            "value": "16675.9613(万元)",
          },
          Object {
            "code": "number of participants",
            "name": "参保人数",
            "value": 79,
          },
          Object {
            "code": "register prov",
            "name": "注册省份",
            "value": "西藏自治区",
          },
        ],
        "type": "company",
      },
      "id": "node-5",
      "shape": "CircleNode",
      "style": Object {
        "dark": "#eee",
        "fontColor": "#3b3b3b",
        "fontSize": 12,
        "nodeSize": 20,
        "primaryColor": "#515151",
      },
      "x": 381,
      "y": 475,
    },
    Object {
      "data": Object {
        "id": "node-6",
        "label": "寇交践某某公司(node-6)",
        "properties": Array [
          Object {
            "code": "name",
            "name": "公司名称",
            "value": "寇交践某某公司",
          },
          Object {
            "code": "companyType",
            "name": "类型",
            "value": "非上市公司",
          },
          Object {
            "code": "industry",
            "name": "行业",
            "value": "互联网行业",
          },
          Object {
            "code": "phone",
            "name": "电话",
            "value": "361482999",
          },
          Object {
            "code": "USCC",
            "name": "统一社会信用代码",
            "value": "9581986256248045",
          },
          Object {
            "code": "registered capital",
            "name": "注册资本",
            "value": "59167.0185(万元)",
          },
          Object {
            "code": "number of participants",
            "name": "参保人数",
            "value": 15,
          },
          Object {
            "code": "register prov",
            "name": "注册省份",
            "value": "湖南省",
          },
        ],
        "type": "company",
      },
      "id": "node-6",
      "shape": "CircleNode",
      "style": Object {
        "dark": "#eee",
        "fontColor": "#3b3b3b",
        "fontSize": 12,
        "nodeSize": 20,
        "primaryColor": "#515151",
      },
      "x": 467,
      "y": 475,
    },
    Object {
      "data": Object {
        "id": "node-7",
        "label": "熊全某某公司(node-7)",
        "properties": Array [
          Object {
            "code": "name",
            "name": "公司名称",
            "value": "熊全某某公司",
          },
          Object {
            "code": "companyType",
            "name": "类型",
            "value": "非上市公司",
          },
          Object {
            "code": "industry",
            "name": "行业",
            "value": "传统行业",
          },
          Object {
            "code": "phone",
            "name": "电话",
            "value": "604129966",
          },
          Object {
            "code": "USCC",
            "name": "统一社会信用代码",
            "value": "7678397415260516",
          },
          Object {
            "code": "registered capital",
            "name": "注册资本",
            "value": "37060.7734(万元)",
          },
          Object {
            "code": "number of participants",
            "name": "参保人数",
            "value": 47,
          },
          Object {
            "code": "register prov",
            "name": "注册省份",
            "value": "西藏自治区",
          },
        ],
        "type": "company",
      },
      "id": "node-7",
      "shape": "CircleNode",
      "style": Object {
        "dark": "#eee",
        "fontColor": "#3b3b3b",
        "fontSize": 12,
        "nodeSize": 20,
        "primaryColor": "#515151",
      },
      "x": 553,
      "y": 475,
    },
    Object {
      "data": Object {
        "id": "node-8",
        "label": "龙胡定某某公司(node-8)",
        "properties": Array [
          Object {
            "code": "name",
            "name": "公司名称",
            "value": "龙胡定某某公司",
          },
          Object {
            "code": "companyType",
            "name": "类型",
            "value": "上市公司",
          },
          Object {
            "code": "industry",
            "name": "行业",
            "value": "互联网行业",
          },
          Object {
            "code": "phone",
            "name": "电话",
            "value": "144790128",
          },
          Object {
            "code": "USCC",
            "name": "统一社会信用代码",
            "value": "9186344430201236",
          },
          Object {
            "code": "registered capital",
            "name": "注册资本",
            "value": "23680.2105(万元)",
          },
          Object {
            "code": "number of participants",
            "name": "参保人数",
            "value": 100,
          },
          Object {
            "code": "register prov",
            "name": "注册省份",
            "value": "海南省",
          },
        ],
        "type": "company",
      },
      "id": "node-8",
      "shape": "CircleNode",
      "style": Object {
        "dark": "#eee",
        "fontColor": "#3b3b3b",
        "fontSize": 12,
        "nodeSize": 20,
        "primaryColor": "#515151",
      },
      "x": 639,
      "y": 475,
    },
    Object {
      "data": Object {
        "id": "node-9",
        "label": "夏崩己阴某某公司(node-9)",
        "properties": Array [
          Object {
            "code": "name",
            "name": "公司名称",
            "value": "夏崩己阴某某公司",
          },
          Object {
            "code": "companyType",
            "name": "类型",
            "value": "非上市公司",
          },
          Object {
            "code": "industry",
            "name": "行业",
            "value": "互联网行业",
          },
          Object {
            "code": "phone",
            "name": "电话",
            "value": "658722341",
          },
          Object {
            "code": "USCC",
            "name": "统一社会信用代码",
            "value": "750002646477931",
          },
          Object {
            "code": "registered capital",
            "name": "注册资本",
            "value": "64985.0703(万元)",
          },
          Object {
            "code": "number of participants",
            "name": "参保人数",
            "value": 60,
          },
          Object {
            "code": "register prov",
            "name": "注册省份",
            "value": "云南省",
          },
        ],
        "type": "company",
      },
      "id": "node-9",
      "shape": "CircleNode",
      "style": Object {
        "dark": "#eee",
        "fontColor": "#3b3b3b",
        "fontSize": 12,
        "nodeSize": 20,
        "primaryColor": "#515151",
      },
      "x": 37,
      "y": 75,
    },
  ],
}
`;
